release_checklist.md

```diff
@@ -1,47 +1,117 @@
-1) Checkout master
-1) `make test`
-1) `bazel test ...:all`
-1) `mkdir build ; cd build ; cmake .. ; make ; make test`
-1) modify `include/libjsonnet.h` to bump the version number but add `-pre1`
-1)
+# Jsonnet release checklist
+
+## Preparation
+
+Before doing the release, make sure that the project is in good state:
+
+* Triage issues – make sure that all issues are labelled.
+* Check for release-blocking bugs. All bugs which results in a wrong result of the evaluation are release-blocking.
+* Go through the open PRs – consider merging any outstanding ones. Do not merge big changes right before the release.
+* Sync google/go-jsonnet and google/jsonnet
+    1) Check out master from both
+    1) Check go-jsonnet/cpp-jsonnet submodule is HEAD of jsonnet tree, if not:
+    1) Inside cpp-jsonnet:
+    1) `git checkout master`
+    1) `git pull`
+    1) `cd ..`
+    1) `git checkout -b release`
+    1) Update the standard library: `go run cmd/dumpstdlibast/dumpstdlibast.go cpp-jsonnet/stdlib/std.jsonnet > astgen/stdast.go`
+    1) `git commit -a -m 'Update cpp-jsonnet'`
+    1) `./tests.sh`  # Have to commit before running this, or it resets the submodule
+    1) Fix any failing tests.
+    1) Send a PR with an updated version, so that all the CI tests are run
+* Check that CI is green.
+* Make sure that the project can be built and the tests pass using all build systems.
+    * C++:
+        1) `make test`
+        1) `bazel test ...:all`
+        1) `mkdir build ; cd build ; cmake .. ; make ; make test`
+    * Go:
+        1) `./tests.sh`
+        2) `bazel test //:go_default_test`
+* Make sure the Python bindings work (build and run tests locally).
+* Go through the commits since the last release and prepare the release notes.
+    * The release notes should have separate sections for the language changes and
+    changes specific to each implementation.
+* Make sure that the stdlib documentation is complete. Check the release notes for any
+additions or changes to stdlib and make sure they are reflected in the documentation.
+* Make sure that you can build the website locally.
+    * `make doc/js/libjsonnet.js`
+    * `tools/scripts/serve_docs.sh`
+    * Check that it works in two different browsers. Make sure that live evaluation
+    in the tutorial works.
+
+All the above points apply to both google/jsonnet and google/go-jsonnet.
+
+## Make a final decision to release
+
+A this point you should be confident that the project is ready for the release.
+Some of subsequent steps leave project are hard to reverse and/or leave the project in
+an inconsistent state, so proceed only if you have time to complete the release process
+in one sitting and fix any unexpected problems.
+
+## Release the C++ version
+
+* Modify `include/libjsonnet.h` to bump the version number but add `-pre1` suffix.
+* Update `test_cmd` golden file version numbers:
+```
+find test_cmd -name '*.cpp' -o -name '*.golang' -o -name '*.stdout' -o -name '*.stderr' | \
```

Comment:
Put this into a script or eliminate the need for that completely by redacting version numbers from golden files.